[TEST-FAILURE] org.elasticsearch.client.documentation.LicensingDocumentationIT fails on release-tests CI

Build URL: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+release-tests/855/console

I have tried to reproduce this locally without any luck, although I was not able to run with `-Dbuild.snapshot=false` locally so this bug might be specific to release builds.

Reproduce command (as lifted from CI failure, locally I removed the `-Dbuild.snapshot=false`):
```
REPRODUCE WITH: ./gradlew :client:rest-high-level:integTestRunner \
  -Dtests.seed=6B866F493C3B7898 \
  -Dtests.class=org.elasticsearch.client.documentation.LicensingDocumentationIT \
  -Dtests.method="testPutLicense" \
  -Dtests.security.manager=true \
  -Dbuild.snapshot=false \
  -Dtests.jvm.argline="-Dbuild.snapshot=false" \
  -Dtests.locale=el-GR \
  -Dtests.timezone=Africa/Khartoum
```

It looks like the test is trying to put an invalid license and expecting ES to return an exception saying the license is valid but the server is returning that the licence is invalid. Maybe the way the test is creating an valid licence might be wrong:
```
  1> [2018-08-02T10:45:24,332][INFO ][o.e.c.d.LicensingDocumentationIT] [testPutLicense] before test
  2> NOTE: leaving temporary files on disk at: /var/lib/jenkins/workspace/elastic+elasticsearch+master+release-tests/client/rest-high-level/build/testrun/integTestRunner/J0/temp/org.elasticsearch.client.documentation.LicensingDocumentationIT_6B866F493C3B7898-001
  2> NOTE: test params are: codec=Asserting(Lucene70): {}, docValues:{}, maxPointsInLeafNode=1403, maxMBSortInHeap=7.8940949346565645, sim=RandomSimilarity(queryNorm=false): {}, locale=el-GR, timezone=Africa/Khartoum
  2> NOTE: Linux 4.4.0-1061-aws amd64/Oracle Corporation 1.8.0_181 (64-bit)/cpus=16,threads=1,free=360450152,total=516947968
  1> [2018-08-02T10:45:24,333][INFO ][o.e.c.d.LicensingDocumentationIT] [testPutLicense] initializing REST clients against [http://[::1]:39337]
  2> NOTE: All tests run in this JVM: [SnapshotClientDocumentationIT, TasksClientDocumentationIT, WatcherDocumentationIT, TasksIT, PingAndInfoIT, IndicesClientDocumentationIT, MigrationDocumentationIT, SearchDocumentationIT, WatcherIT, CRUDDocumentationIT, ClusterClientDocumentationIT, IngestClientIT, IndicesClientIT, SnapshotIT, BulkProcessorIT, IngestClientDocumentationIT, RankEvalIT, StoredScriptsIT, StoredScriptsDocumentationIT, BulkProcessorRetryIT, ClusterClientIT, CrudIT, SearchIT, LicensingDocumentationIT]
  1> [2018-08-02T10:45:24,352][INFO ][o.e.c.d.LicensingDocumentationIT] [testPutLicense] after test
FAILURE 0.03s | LicensingDocumentationIT.testPutLicense <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: expected:<invalid> but was:<valid>
   > 	at __randomizedtesting.SeedInfo.seed([6B866F493C3B7898:8EDA9690B7D4DA9F]:0)
   > 	at org.elasticsearch.client.documentation.LicensingDocumentationIT.testPutLicense(LicensingDocumentationIT.java:68)
   > 	at java.lang.Thread.run(Thread.java:748)
```

--- a/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/LicensingDocumentationIT.java
+++ b/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/LicensingDocumentationIT.java
@@ -19,7 +19,7 @@
 
 package org.elasticsearch.client.documentation;
 
-import org.apache.lucene.util.LuceneTestCase.AwaitsFix;
+import org.elasticsearch.Build;
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.action.LatchedActionListener;
 import org.elasticsearch.client.ESRestHighLevelClientTestCase;
@@ -41,10 +41,10 @@ import static org.hamcrest.Matchers.startsWith;
  * Documentation for Licensing APIs in the high level java client.
  * Code wrapped in {@code tag} and {@code end} tags is included in the docs.
  */
-@AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/32580")
 public class LicensingDocumentationIT extends ESRestHighLevelClientTestCase {
 
     public void testPutLicense() throws Exception {
+        assumeTrue("License is only valid when tested against snapshot/test builds", Build.CURRENT.isSnapshot());
         RestHighLevelClient client = highLevelClient();
         String license = "{\"license\": {\"uid\":\"893361dc-9749-4997-93cb-802e3d7fa4a8\",\"type\":\"gold\"," +
             "\"issue_date_in_millis\":1411948800000,\"expiry_date_in_millis\":1914278399999,\"max_nodes\":1,\"issued_to\":\"issued_to\"," +

